pymodbus/pdu/pdu.py

```diff
@@ -128,7 +127,21 @@ def __init__(
         super().__init__(transaction_id=transaction, dev_id=device_id)
         self.function_code = function_code | 0x80
         self.exception_code = exception_code
-        Log.error(f"Exception response {self.function_code} / {self.exception_code}")
+
+    def __str__(self) -> str:
+        """Build a representation of an exception response."""
+        return (
+            f"{self.__class__.__name__}("
```

Comment:
This is far too complicated !!! you are adding a lot of fields that are not used and a lot more that are only used with some function codes.

It is a representation of the "exception response" NOT every possible response.